executor/executor.go

```diff
@@ -175,6 +175,17 @@ type Executor interface {
 	newFirstChunk() *chunk.Chunk
 }
 
+// Next is a wrapper function on e.Next(), it handles some common codes.
+func Next(ctx context.Context, e Executor, req *chunk.RecordBatch) error {
+	select {
```

Comment:
what's the performance regression for large queries which require to call a huge amount of Next() function?